subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/MavenConversionIntegrationTest.groovy

```diff
@@ -649,6 +649,36 @@ Root project 'webinar-parent'
         dsl.getBuildFile().text.contains(TextUtil.toPlatformLineSeparators(mavenLocalRepoBlock))
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/20981")
+    def "escapeSingleQuotes"() {
+        def dsl = dslFixtureFor(scriptDsl)
+
+        when:
+        run 'init', '--dsl', scriptDsl.id as String
+
+        then:
+        dsl.assertGradleFilesGenerated()
+
+        def isGroovy = scriptDsl == BuildInitDsl.GROOVY
+        def descriptionPropertyAssignment = (isGroovy ? 'description = \'That\\\'s it' : 'description = "That\'s it"')
+        dsl.getBuildFile().text.contains(descriptionPropertyAssignment)
+    }
+
+    @Issue("https://github.com/gradle/gradle/issues/20981")
+    def "escapeDoubleQuotes"() {
+        def dsl = dslFixtureFor(scriptDsl)
+
+        when:
+        run 'init', '--dsl', scriptDsl.id as String
+
+        then:
+        dsl.assertGradleFilesGenerated()
+
+        def isGroovy = scriptDsl == BuildInitDsl.GROOVY
+        def descriptionPropertyAssignment = (isGroovy ? 'description = \'"Quoted description"\'' : 'description = "\\"Quoted description\\"')
+        dsl.getBuildFile().text.contains(descriptionPropertyAssignment)
```

Comment:
Here and L664, suggest using this check instead:

```
dsl.getBuildFile().text.readLines().contains(descriptionPropertyAssignment)
```

That would show the Kotlin test for escaping double-quoted strings is missing the closing quote.